pkg/util/kubernetes/apiserver/resourcetypes_test.go

```diff
@@ -0,0 +1,258 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package apiserver
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestInitializeGlobalResourceTypeCache(t *testing.T) {
+	resetCache()
+
+	client := fakeclientset.NewClientset()
+	fakeDiscoveryClient := client.Discovery().(*fakediscovery.FakeDiscovery)
+	fakeDiscoveryClient.Resources = []*v1.APIResourceList{}
+
+	tests := []struct {
+		name      string
+		discovery discovery.DiscoveryInterface
+		wantErr   bool
+	}{
+		{
+			name:      "Initialize cache successfully",
+			discovery: fakeDiscoveryClient,
+			wantErr:   false,
+		},
+		{
+			name:      "Re-initialization does nothing",
+			discovery: fakeDiscoveryClient,
+			wantErr:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := InitializeGlobalResourceTypeCache(tt.discovery)
+			if tt.wantErr {
```

Comment:
In both of these tests, `wantErr` is `false` - do you need this conditional? / are there other tests cases that should be added to test for `IsGroupDiscoveryFailedError`/other errors?

if no other tests need to be added, I think this would make more sense as a non-table test (since you want these to be done sequentially anyway I think? [initialization and then re-initialization])

Reply:
As a further test that re-initializing doesn't change the cache, can you add resources to `fakeDiscoveryClient.Resources` before re-initializing and check that the cache is unchanged?